internal/common/db.go

```diff
@@ -101,6 +107,27 @@ func (c *Cluster) Payload() any {
 	return &c.Cluster
 }
 
+func (c *Cluster) GetMirrorRegistryConfiguration() (*v1beta1.MirrorRegistryConfiguration, error) {
+	if c.MirrorRegistryConfiguration == nil {
+		return nil, nil
+	}
+	return convertBytesToMirrorRegistryConfig(c.MirrorRegistryConfiguration)
+}
+
+func (c *Cluster) SetMirrorRegistryConfiguration(configuration *v1beta1.MirrorRegistryConfiguration) error {
+	if configuration == nil {
+		return nil
```

Comment:
This makes it look like there's no way to explicitly remove the mirror configuration. Do we want to deal with that scenario?

Reply:
Fixed that in last commit